src/test/java/com/github/neatlife/jframework/util/PrecisionUtilTest.java

```diff
@@ -0,0 +1,137 @@
+        Assert.assertEquals(new BigDecimal("10"),
+                PrecisionUtil.multiply(null, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.multiply(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalDivide() {
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0"),
+                PrecisionUtil.divide(null, BigDecimal.TEN, 1));
+        Assert.assertEquals(new BigDecimal("1.0"),
+                PrecisionUtil.divide(BigDecimal.ONE, null, 1));
+    }
+
+    @Test
+    public void testStringAdd() {
+        Assert.assertEquals("1", PrecisionUtil.add("1", null));
+        Assert.assertEquals("2", PrecisionUtil.add(null, "2"));
+    }
+
+    @Test
+    public void testStringSubtract() {
+        Assert.assertEquals("1", PrecisionUtil.subtract("1", null));
+        Assert.assertEquals("-2", PrecisionUtil.subtract(null, "2"));
+    }
+
+    @Test
+    public void testStringMultiply() {
+        Assert.assertEquals("1", PrecisionUtil.multiply("1", null));
+        Assert.assertEquals("2", PrecisionUtil.multiply(null, "2"));
+    }
+
+    @Test
+    public void testStringDivide() {
+        Assert.assertEquals("1.00", PrecisionUtil.divide("1", null));
+        Assert.assertEquals("0", PrecisionUtil.divide(null, "2"));
+    }
+
+    @Test
+    public void testBigDecimalSum() {
+        Assert.assertEquals(new BigDecimal("11"), PrecisionUtil.sum(null,
+                BigDecimal.ZERO, BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.sum(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testStringSum() {
+        Assert.assertEquals("1", PrecisionUtil.sum("1", null));
+        Assert.assertEquals("6", PrecisionUtil.sum(null, "1", "2", "3"));
+    }
+
+    @Test
+    public void testBigDecimalAvg() {
+        Assert.assertEquals(new BigDecimal("3.67"), PrecisionUtil.avg(
+                BigDecimal.ZERO, BigDecimal.ONE, BigDecimal.TEN));
+    }
+
+    @Test
+    public void testStringAvg() {
+        Assert.assertEquals("2.00", PrecisionUtil.avg("1", "2", "3"));
+    }
+
+    @Test
+    public void testBigDecimalMaxArr() {
+        Assert.assertEquals(BigDecimal.TEN, PrecisionUtil.maxArr(
+                BigDecimal.ZERO, BigDecimal.TEN, BigDecimal.ONE));
+    }
+
+    @Test
+    public void testBigDecimalMinArr() {
+        Assert.assertEquals(BigDecimal.ZERO, PrecisionUtil.minArr(
+                BigDecimal.ONE, BigDecimal.ZERO, BigDecimal.TEN));
+    }
+
+    @Test
+    public void testStringMaxArr() {
+        Assert.assertNull(PrecisionUtil.max(null, "1", "2", "3"));
+        Assert.assertEquals("4", PrecisionUtil.maxArr("2", "4", "1", "3"));
```

Comment:
Add a test where `null` is not the first element, and one with an empty string(s) included

Reply:
Done